Add new heading and description for partitionIndex 🤷‍♂️. (#137)
apple/swift-algorithms: README.md

- [`firstNonNil(_:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/FirstNonNil.md): Returns the first non-`nil` result from transforming a sequence's elements.
- [`indexed()`](https://github.com/apple/swift-algorithms/blob/main/Guides/Indexed.md): Iterate over tuples of a collection's indices and elements. 
- [`interspersed(with:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Intersperse.md): Place a value between every two elements of a sequence.

> - [`partitioningIndex(where:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Partition.md)`: Returns the starting index of the partition of a collection that matches a predicate.

Reviewer: ```suggestion
- [`partitioningIndex(where:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Partition.md): Returns the starting index of the partition of a collection that matches a predicate.
```

Syntax error.

> - [`partitioningIndex(where:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Partition.md)`: Splits a collection into two partitions according to a predicate and returns the index of the partition.

Reviewer 2: @LemonSpike Sorry, hadn't actually read the text here! :facepalm:

`partitioningIndex` doesn't modify the collection, so let's go with something like this instead:

```suggestion
- [`partitioningIndex(where:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Partition.md)`: Returns the starting index of the partition of a collection that matches a predicate.
```

- [`split(maxSplits:omittingEmptySubsequences:whereSeparator)`, `split(separator:maxSplits:omittingEmptySubsequences)`](https://github.com/apple/swift-algorithms/blob/main/Guides/LazySplit.md): Lazy versions of the Standard Library's eager operations that split sequences and collections into subsequences separated by the specified separator element.
- [`windows(ofCount:)`](https://github.com/apple/swift-algorithms/blob/main/Guides/Windows.md): Breaks a collection into overlapping subsequences where elements are slices from the original collection.

> #### Binary Search Partitions

Reviewer: If this deserves its own heading, then it should follow the same capitalization and grammatical phrasing as other headings, and it should be above "Other useful operations"; otherwise, it should be added to the latter section in alphabetical order.

Author: Thanks for the feedback @xwu! I was following issue #131 " It needs to be in a different category, though."

Reviewer 2: Let's add it to "Other useful operations" for now, thanks!
